torch/csrc/distributed/c10d/TCPStore.cpp

```diff
@@ -396,6 +396,18 @@ void TCPStore::validate(void) {
   buffer.flush();
 }
 
+void TCPStore::_splitSet(
```

Comment:
Can we actually make this into test by inherit TCPStore and add it to the child class instead of adding it to TCPStore itself?

Reply:
IMO that would require a refactor of TCPStore because the TCPClient object is a Private member of TCPStore and can only be accessed from the TCPStore object.

Reply:
i think @fduwjj means, can you make a TestTCPStore class in TcpStoreTest.cpp, and just implement the extra method on that?

Reply:
I have tried that attempt, but the thing is, the subclass cannot access the private member of the superclass, so the _splitSet function in the subclass won't be able to use the buffer

Reply:
ok. could probably change the base class member to protected, add your test class as a friend class, or just copy some code into the test class.

Or, maybe you don't even need a tcpstore at all- just create a raw tcpclient obj in your test case, and then write the 'split_set' as a standalone helper function that uses tcpclient?

Reply:
Totally agree with the second solution (do not create a TCPStore object in the client thread but directly copy paste the socket connect and send logic). But I would prefer to merging this PR first then make that change in a follow-up to speed up since it's not harming. WDYT? @wconstab @fduwjj

Reply:
ok. i was thinking about saying that you should just reverse the order, land the next refactor PR first, then land this one.  But i think its worth landing the fix now and its not that bad to add the _split_set, as long as you promise to clean it up soon.

Reply:
@wconstab @fduwjj for the follow-up there're several options I can come up with, let me know your opinions:
1. in test's TCP store client thread, do not use any `TCPStore`, `SendBuffer`, or `TCPClient` because they either cannot be used outside of `TCPStore.cpp` or encapsulate private members from users. Instead, use the `Socket` to perform the exact same thing as in those classes. The downside is making the test quite verbose (but we can hide them in test util files).

2. change some members to Protected and make the test class a friend class. The downside is I don't know the consequence of making the underlying networking components `Protected`. Can there be any security issue? Who would be the expert we can ask?

3. Refactor the structure of `TCPStore.hpp` and `TCPStoreBackend.hpp`. Currently, `TCPClient` and `SendBuffer` can only have incomplete declaration in `TCPStore.hpp` because `QueryType` is defined in `TCPStoreBackend.hpp` and `TCPStoreBackend.hpp` include `TCPStore.hpp`. This forbids them from being exposed outside of `TCPStore.cpp`. The question is, do we want to refactor them? Any benefit? A single test may not be a good justification.

Reply:
I think #2 should be fine? What security issue can you think of?

Reply:
If another class A inherits `TCPStore`, the class A or an instance of A can access a `TCPStore`'s buffer and read/write on it. I assume the original design decision (making them `private`) is to avoid this.

Reply:
Well, what's the problem for that? If it's a child of `TCPStore` then it's ok to access the buffer?